src/lib/badge/badge.ts

```diff
@@ -0,0 +1,180 @@
+ */
+
+import {Directive, Input, ElementRef, Inject, Optional, NgZone} from '@angular/core';
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+import {ThemePalette} from '@angular/material/core';
+import {AriaDescriber} from '@angular/cdk/a11y';
+import {DOCUMENT} from '@angular/common';
+
+let nextId = 0;
+
+export type PositionTypes =
+  'above after' |
+  'below after' |
+  'above before' |
+  'above after';
+
+export type BadgeSize =
+  'small' |
+  'medium' |
+  'large';
+
+/** Directive to display a text badge. */
+@Directive({
+  selector: '[matBadge]',
+  inputs: ['matBadgeColor', 'matBadgeOverlap', 'matBadgePosition'],
+  host: {
+    'class': 'mat-badge',
+    '[class.mat-badge-overlap]': '_overlap',
+    '[class.mat-badge-above]': '_isAbove',
+    '[class.mat-badge-below]': '!_isAbove',
+    '[class.mat-badge-before]': '!_isAfter',
+    '[class.mat-badge-after]': '_isAfter',
+    '[class.mat-badge-small]': 'matBadgeSize === "small"',
+    '[class.mat-badge-medium]': 'matBadgeSize === "medium"',
+    '[class.mat-badge-large]': 'matBadgeSize === "large"',
+    '[class.mat-badge-hidden]': 'matBadgeHidden',
+  },
+})
+export class MatBadge {
+
+  /** The color of the badge. Can be `primary`, `accent`, or `warn`. */
+  @Input('matBadgeColor')
+  get color(): ThemePalette { return this._color; }
+  set color(value: ThemePalette) {
+    this._color = value;
+    this._setColor(value);
+  }
+  private _color: ThemePalette = 'primary';
+
+  /** Whether the badge should overlap its contents or not */
+  @Input('matBadgeOverlap')
+  get overlap(): boolean { return this._overlap; }
+  set overlap(val: boolean) {
+    this._overlap = coerceBooleanProperty(val);
+  }
+  private _overlap: boolean = true;
+
+  /**
+   * Position the badge should reside.
+   * Accepts any combination of 'above'|'below' and 'before'|'after'
+   */
+  @Input('matBadgePosition')
+  get position(): PositionTypes { return this._position; }
+  set position(val: PositionTypes) {
+    this._position = val;
+    this._isAbove = val.indexOf('below') === -1;
+    this._isAfter = val.indexOf('before') === -1;
+  }
+  private _position: PositionTypes = 'above after';
+
+  /** The content for the badge */
+  @Input('matBadge')
+  get content(): string { return this._content; }
+  set content(val: string) {
+    this._content = val;
+    this._updateTextContent();
+  }
+  private _content: string;
+
+  /** Message used to describe the decorated element via aria-describedby */
+  @Input('matBadgeDescription')
+  get description(): string { return this._description; }
+  set description(val: string) {
+    this._updateHostAriaDescription(val, this._description);
+    this._description = val;
+  }
+  private _description: string;
+
+  /** Size of the badge. Can be 'small' , 'medium', or 'large'. */
+  @Input() matBadgeSize: BadgeSize = 'medium';
+
+  /** Whether the badge is hidden. */
+  @Input()
+  set matBadgeHidden(val: boolean) {
```

Comment:
`getters` should come before than `setters`. See #9471.

Reply:
Yep, it should be `hidden` with an `@Input` alias